.github/workflows/build-image.yaml

```diff
@@ -0,0 +1,44 @@
+name: Create and publish a Docker image
+
+on:
+  push:
```

Comment:
According to the metadata action template, I would expect:
```
  push:
    branches:
      - 'master'
    tags:
      - 'v*'
  pull_request:
    branches:
      - 'master'
 ```
here.

Not sure if github has a `default_branch` variable which would be useful here.